Source/problems/Castro_bc_fill_nd.cpp

```diff
@@ -109,17 +110,21 @@ void ca_statefill(Box const& bx, FArrayBox& data,
         (bcr[URHO].hi(0) == EXT_DIR &&           // xr, yr, zr corner
          (bcr[URHO].hi(1) == EXT_DIR || bcr[URHO].hi(2) == EXT_DIR)) ||
         (bcr[URHO].hi(1) == EXT_DIR && bcr[URHO].hi(2) == EXT_DIR)) {
-      amrex::Error("Error: external boundaries meeting at a corner not supported");
+        amrex::Error("Error: external boundaries meeting at a corner not supported");
     }
 #endif
 
     if (numcomp == 1) {
 
+#ifndef CXX_MODEL_PARSER
 #pragma gpu box(bx)
         ext_denfill(AMREX_INT_ANYD(bx.loVect()), AMREX_INT_ANYD(bx.hiVect()),
                     BL_TO_FORTRAN_N_ANYD(data, dcomp),
                     AMREX_INT_ANYD(geom.Domain().loVect()), AMREX_INT_ANYD(geom.Domain().hiVect()),
                     AMREX_REAL_ANYD(geom.CellSize()), AMREX_REAL_ANYD(geom.ProbLo()), time, bc_f);
+#else
+        ext_denfill_c(bx, data.array(), dcomp, geom, bcr[dcomp], time);
```

Comment:
The `array()` method has an overload that accepts the component, so you could do `data.array(dcomp)` here if you wanted the denfill routine to look more like it does in Fortran now.